treeherder/auth/backends.py

```diff
@@ -0,0 +1,148 @@
+import logging
+import re
+from hashlib import sha1
+
+from django.contrib.auth.models import User
+from django.core.exceptions import ObjectDoesNotExist
+from django.db.models import Q
+from taskcluster.sync import Auth
+from taskcluster.utils import scope_match
+
+logger = logging.getLogger(__name__)
+
+
+class TaskclusterAuthBackend(object):
+    """
+        result of tc_auth.authenticateHawk has the form:
+
+        {'status': 'auth-success',
+         'scopes': ['assume:mozilla-group:ateam',
+                    'assume:mozilla-group:vpn_treeherder',
+                    'assume:mozilla-user:biped@mozilla.com',
+                    'assume:mozillians-user:biped',
+                    ...
+                    'assume:project-admin:ateam',
+                    'assume:project-admin:treeherder',
+                    'assume:project:ateam:*',
+                    'assume:project:treeherder:*',
+                    'assume:worker-id:*',
+                    'secrets:set:project/treeherder/*'],
+         'scheme': 'hawk',
+         'clientId': 'mozilla-ldap/biped@mozilla.com',
+         'expires': '2016-10-31T17:40:45.692Z'}
+    """
+
+    def _get_scope_value(self, result, scope_prefix):
+        for scope in result["scopes"]:
+            if scope.startswith(scope_prefix):
+                return scope[len(scope_prefix):]
+        return None
+
+    def _get_email(self, result):
+        """
+        Get the user's email from the mozilla-user scope
+
+        """
+        email = self._get_scope_value(result, "assume:mozilla-user:")
+
+        if not email or email == "*":
+            # try finding the email in the clientId.  It can then be used to
```

Comment:
To clarify, I mean that I think we shouldn't fall back to using `result["clientId"]`, since Dustin said it was insecure?